kong/pdk/ctx.lua

```diff
@@ -86,18 +88,18 @@ local _CTX_CORE_KEY = {}
 --
 --   kong.log(value) -- "hello world"
 -- end
-local function new(self)
-  local _CTX = {
-    -- those would be visible on the *.ctx namespace for now
-    -- TODO: hide them in a private table shared between this
-    -- module and the global.lua one
-    keys = setmetatable({}, { __mode = "v" }),
-  }
 
 
+local function new(self)
+  local _CTX = {}
   local _ctx_mt = {}
 
 
+  function _ctx_mt.__newindex(t, k, v)
```

Comment:
Better to revert back to the previous `_ctx_mt.set` suggestion. Using the `__newindex` metamethod isn't ideal here...

Reply:
Reverted, but now I have some errors :

```
[error] 26220#0: *16 lua entry thread aborted: runtime error: ...4b3d6e9e6e088fd99/luarocks/share/lua/5.1/kong/global.lua:58: attempt to call field 'set' (a nil value)
```

I am lost, I don't understand why we should make `.set` on `_ ctx_mt` and not on `_CTX`,  where i made the mistake...

Reply:
The goal is to avoid redefining copies of the function when multiple instances of the ctx PDK module are created. But it would require updates to the `__index` function... If you aren't familiar with Lua, then nevermind, let's revert back to adding the function to the `_CTX` table, we can take care of those other details later.